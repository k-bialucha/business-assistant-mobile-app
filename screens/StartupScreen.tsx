/**
 * Startup module to check if user's data are saved in local storage, then log in automatically otherwise
 * set in redux didTryLogin to true
 */

import React, { useEffect } from 'react';
import { ActivityIndicator } from 'react-native';

import { StyledView } from './StartupScreen.style';

const StartupScreen = () => {
  // try log in when Components lodaded
  useEffect(() => {}, []);

  return (
    <StyledView>
      <ActivityIndicator size="large" color="black" />
    </StyledView>
  );
};

export default StartupScreen;
