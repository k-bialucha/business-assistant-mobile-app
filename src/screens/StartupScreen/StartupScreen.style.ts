import styled from 'styled-components/native';

import Colors from '../../theme/Colors';

const StyledView = styled.View`
  flex: 1;
  align-items: center;
  justify-content: center;
  background-color: ${Colors.navyBlue};
`;

export { StyledView };
