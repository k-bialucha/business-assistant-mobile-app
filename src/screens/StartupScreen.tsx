/**
 * Startup module to check if user's data are saved in local storage, then log in automatically otherwise
 * set in redux didTryLogin to true
 */

import React, { useEffect } from 'react';
import { ActivityIndicator } from 'react-native';

import { useDispatch } from 'react-redux';

import { tryAutoLogin } from '../redux/auth/actions';

import { StyledView } from './StartupScreen.style';

const StartupScreen = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(tryAutoLogin());
  }, [dispatch]);

  return (
    <StyledView>
      <ActivityIndicator size="large" color="white" />
    </StyledView>
  );
};

export default StartupScreen;
