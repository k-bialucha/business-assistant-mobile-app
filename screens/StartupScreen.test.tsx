import React from 'react';

import { shallow } from 'enzyme';

import StartupScreen from './StartupScreen';

describe('<StartupScreen />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<StartupScreen />);

    expect(tree).toMatchSnapshot();
  });
});
