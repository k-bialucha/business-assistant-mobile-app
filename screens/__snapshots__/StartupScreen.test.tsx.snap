// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartupScreen /> matches the snapshot 1`] = `
<Styled(View)>
  <ActivityIndicator
    animating={true}
    color="black"
    hidesWhenStopped={true}
    size="large"
  />
</Styled(View)>
`;
